catalog/opensearch/catalog-opensearch-source/src/test/java/ddf/catalog/source/opensearch/TestOpenSearchParserImpl.java

```diff
@@ -0,0 +1,631 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source.opensearch;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+
+import org.apache.cxf.jaxrs.client.WebClient;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+import org.junit.Before;
+import org.junit.Test;
+import org.opengis.filter.Filter;
+import org.opengis.filter.sort.SortBy;
+import org.opengis.filter.sort.SortOrder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Result;
+import ddf.catalog.filter.impl.SortByImpl;
+import ddf.catalog.impl.filter.SpatialDistanceFilter;
+import ddf.catalog.impl.filter.SpatialFilter;
+import ddf.catalog.impl.filter.TemporalFilter;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+
+public class TestOpenSearchParserImpl {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestOpenSearchParserImpl.class);
+
+    private static final StringBuilder URL = new StringBuilder(
+            "http://localhost:8080/services/catalog/query");
+
+    private static final String MAX_RESULTS = "2000";
+
+    private static final String TIMEOUT = "30000";
+
+    private static final String DESCENDING_TEMPORAL_SORT = "date%3Adesc";
+
+    private OpenSearchParserImpl openSearchParserImpl;
+
+    @Before
+    public void setUp() {
+        openSearchParserImpl = new OpenSearchParserImpl();
+    }
+
+    @Test
+    public void populateSpatialDistanceFilterBbox() throws Exception {
```

Comment:
In this case though, it looks like most of the logic can be kicked out into a helper method that takes the `SpatialDistanceFilter` as a parameter since it's the only difference